.github/workflows/cpplint-scan.yml

```diff
@@ -0,0 +1,60 @@
+name: "Cpplint Scan"
+
+on:
+  push:
+    branches: [master, devel]
+  pull_request:
+    # The branches below must be a subset of the branches above
+    branches: [master, devel]
+
+jobs:
+  cppcheck:
+    name: Cpplint
+    runs-on: ubuntu-latest
+    permissions:
+      actions: read
+      contents: read
+      security-events: write
+
+    steps:
+      - name: "Checkout F´ Repository"
+        uses: actions/checkout@v3
+        with:
+          fetch-depth: 0
+      - uses: ./.github/actions/setup
+
+      - name: Install cpplint
+        run: pip install cpplint
+
+      - name: Install xsltproc
+        run: sudo apt-get install xsltproc -y
+
+      - name: Install sarif tool
+        run: npm install @microsoft/sarif-multitool
+
+      - name: Run cpplint & export output to cppcheck format
+        run: cpplint --counting=detailed --quiet --recursive . 2>&1 | python3 .github/scripts/cpplint_to_cppcheckxml.py &> cpplint_cppcheck_result.xml
+
+      - name: Convert cpplint results to SARIF
+        run: npx "@microsoft/sarif-multitool" convert "cpplint_cppcheck_result.xml" --tool "CppCheck" --output "cpplint_cppcheck_result.sarif" --force
+
+      - name: Convert cpplint results to Markdown & Integrate them in the workflow summary
+        run: xsltproc .github/scripts/cpplint-xml2text.xslt cpplint_cppcheck_result.xml | tee $GITHUB_STEP_SUMMARY cpplint_cppcheck_result.txt
+
+      - name: Upload SARIF file to GitHub Code Scanning Alerts
+        uses: github/codeql-action/upload-sarif@v2
+        with:
+          sarif_file: ${{ github.workspace }}/cpplint_cppcheck_result.sarif
+          checkout_path: ${{ github.workspace }}/source
```

Comment:
I should have caught that sooner but the SARIF upload  is currently failing, the last `/source` is probably from another repo structure and simply needs to be removed here. But I’m not sure what checkout_path is for in this context, I’ll let you find that out.

Reply:
OK, I'll investigate next week!

Reply:
I removed this line, and also the same in the Cppcheck workflow.